services/rsources/mapper.go

```diff
@@ -0,0 +1,66 @@
+package rsources
+
+func statusFromQueryResult(jobRunId string, statMap map[JobTargetKey]Stats) JobStatus {
+	status := JobStatus{
+		ID: jobRunId,
+	}
+
+	taskRunIdIndex := make(map[string]int)           // task run id -> index
+	sourceIdIndex := make(map[string]map[string]int) // sourceID -> taskRunId -> index
+
+	for key, stat := range statMap {
+
+		var idx int
+		if _, ok := taskRunIdIndex[key.TaskRunId]; !ok {
+			idx = len(taskRunIdIndex)
+			status.TasksStatus = append(status.TasksStatus, TaskStatus{
+				ID: key.TaskRunId,
+			})
+			taskRunIdIndex[key.TaskRunId] = idx
+		} else {
+			idx = taskRunIdIndex[key.TaskRunId]
+		}
+
+		var source_idx int
+		if _, ok := sourceIdIndex[key.SourceId]; !ok {
+			source_idx = len(status.TasksStatus[idx].SourcesStatus)
```

Comment:
[minor] Avoid python case. (Didn't the linter complaint)
```suggestion
		var sourceIDx int
		if _, ok := sourceIdIndex[key.SourceId]; !ok {
			sourceIDx = len(status.TasksStatus[idx].SourcesStatus)
```